package.json

```diff
@@ -83,7 +83,7 @@
     "array-includes": "^3.0.3",
     "array.prototype.flat": "^1.2.1",
     "contains-path": "^0.1.0",
-    "debug": "^2.6.9",
+    "debug": "^2.6.9 || ^4.1.1",
```

Comment:
this doesn't actually keep the same node support; `npm install` will install debug v4, which means that it will force users to install a top-level v2 and rely on deduping to be able to get back their support. thus, this is a breaking change.

Reply:
The recommendation by NPM is to enable `engine-strict` option which will force v2 be installed on older node versions since debug v4 requires node v8+.
I thought this is a nice way to allow newer versions of the dependency while still having solution for older node versions.

Reply:
That's only something the top-level app can do - meaning, if the consumers of this package have to do *any work whatsoever* to keep the plugin working, then it's a breaking change, and that includes setting `engine-strict` to true (which would be a terrible idea, since "engines" is purely advisory, and should never block installs).

What's the point of a newer version tho? What do we get in v4 or v3 that we don't get in v2?

Reply:
Fair enough.

The reasoning is just keeping up with latest dependencies and allow deduping.

Reply:
Linting is a dev dep, deduping doesn’t matter; as for keeping up with the latest, that only matters if there’s something that we’re missing out on. Otherwise, it’s just capriciously breaking consumers for no good reason.